processor/processor.go

```diff
@@ -1581,6 +1604,26 @@ func (proc *Handle) processJobsForDest(partition string, subJobs subJob) *transf
 		proc.logger.Debug("[Processor] Total jobs written to event_schema: ", len(eventSchemaJobs))
 	}
 
+	if len(archivalJobs) > 0 {
+		err := misc.RetryWithNotify(
+			context.Background(),
+			proc.jobsDBCommandTimeout,
+			proc.jobdDBMaxRetries,
+			func(ctx context.Context) error {
+				return proc.archivalDB.WithStoreSafeTx(
+					ctx,
+					func(tx jobsdb.StoreSafeTx) error {
+						return proc.archivalDB.StoreInTx(ctx, tx, archivalJobs)
+					},
+				)
+			}, proc.sendRetryStoreStats)
+		if err != nil {
+			proc.logger.Errorf("Store into archival table failed with error: %v", err)
+			proc.logger.Errorf("archival jobs: %v", archivalJobs)
+		}
```

Comment:
We need to make sure that jobs are always archived, or not marked as processed.

With this approach (only logging the error), it is possible to miss some jobs, right?

Reply:
There should have been a panic here which I missed adding. Added that now and the jobs would be retried

Reply:
In this case, we accept that our archivals will contain duplicates right?

Reply:
Assuming that a replay task will send events by prefixing the event's `messageID` with `taskRunId`, processor's deduplication logic can take care of this kind of duplication.

Reply:
This would even be the case right now because GW also might contain duplicates right because of retries? And as Aris mentioned the Dedup logic should cover the same